src/middleware.ts

```diff
@@ -12,12 +12,14 @@ import { parse } from '@polka/url'
 import { lookup } from 'mrmime'
 import { statSync, createReadStream, Stats } from 'node:fs'
 import type { Connect } from 'vite'
+import fs from 'fs-extra'
 import type {
   IncomingMessage,
   OutgoingHttpHeaders,
   ServerResponse
 } from 'node:http'
 import { resolve } from 'node:path'
+import type { SimpleTarget } from './utils'
 
```

Comment:
I think the code would be simplified if the type of `fileMap` is changed to `Map<string, { src: string, transform: (contents: string, filename: string) => string }>` from `Map<string, string>`.